Modify some IF statements syntax to meet WordPress Coding Standards

ihorvorotnov opened the pull request:
**Fixed and updated, replacing messed  and closed #51.**

> After studying the code I've spotted some incorrect usage of standard (curly braces) and alternative syntax for IF statements. As per WordPress PHP Coding Standards, alternative syntax should be used for complex chunks of code, mostly in templates where we mix PHP and HTML. For other cases, especially one-liners, curly braces should be used.

> Or, at the very least, we should have a system. If we obey to use alternative syntax when mixing PHP and HTML, let's do it everywhere (see changes in template-parts/content-single.php and template-parts/content.php). And for one-liners it's better to have braces (see changes in inc/template-tags.php for example).

> Check WordPress PHP Coding Standards here.

karmatosed commented:
I'm not sure the standards prevent or suggest you shouldn't use endif. Can you point to where exactly that's outlined?

ihorvorotnov commented:
@karmatosed here's direct link to WPCS section https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#brace-style

The braces are being discussed a lot over years, for example @pippinsplugins has some good points [on curly braces in templates](https://pippinsplugins.com/please-do-not-use-curly-brackets-in-template-files/) etc.

However, we have WP Coding Standards and have to follow them. And as I wrote in the beginning, using any of the styles is allowed (though one is more readable for templates), but we need to follow one convention, not to mix the styles here and there. 

For example, all over the code if there's any HTML in IF, we use `if/endif`. But in `template-parts/content.php:22` we've got curly braces. 

In `template-parts/content-single.php:37` (and some other places) there's a one-liner `if/endif` which is definitely not necessary.

Also, in `sidebar-content-bottom.php:12` braces were missing, which is not recommended (but overused by many devs).

karmatosed commented:
There isn't anything in the documentation about only doing one method. For example if you view _s the same happens. I'm flagging for a second opinion, but as it's not declaring this absolute in the coding standards, I don't see a requirement for it beyond personal preference. @iamtakashi what is your opinion?

Cleaning up isn't bad as an idea. I'm not sure though that things haven't been done for a reason or legacy issues from _s. If the later is the case then fixing upstream would be right.

ihorvorotnov commented:
Well, the recommendation in docs is really fuzzy, true (and I believe it should be more strict).

`sidebar-content-bottom.php`, line 12:

```
if ( ! is_active_sidebar( 'sidebar-2' ) && ! is_active_sidebar( 'sidebar-3' ) )
    return;
```

However, WPCS says exactly **Braces should always be used, even when they are not required**. At the same time, in `comments.php`, line 18:

```
if ( post_password_required() ) {
    return;
}
```

Which is correct.

Another example, `image.php` on line 49:

```
<?php if ( has_excerpt() ) : ?>
    <div class="entry-caption">
        <?php the_excerpt(); ?>
    </div><!-- .entry-caption -->
<?php endif; ?>
```

which is correct, however, in `content.php` on line 22 we see:

```
<?php if ( has_excerpt() ) { ?>
    <div class="entry-summary">
        <?php the_excerpt(); ?>
    </div><!-- .entry-summary -->
<?php } ?>
```

exactly the same piece of code, but we have braces instead of `if/endif`. It's not a personal preference, it's a mess :)

My point is, default theme is a source for learning and best practices for new users. Many of them just started to learn PHP and are not aware of WPCS, PSR etc. They should not wonder why in one place it's `if(){ ... }` and in another one - `if() ... endif;` doing exactly the same.

This PR sets a kind of _rule of thumb_ - use curly braces for PHP, use `if/endif` for PHP mixed with HTML (because it's easier to read). And it plays well with current WPCS recommendation.

I've put these examples to illustrate the problem and [tweeted some Core devs](https://twitter.com/ihorvorotnov/status/641026242226049024) to hear their opinion. You can ask some of them too, if you don't mind, so we have a better picture.

nacin commented:
I was meticulous in my review of Twenty Ten, Eleven, and Twelve. In more recent themes, I have paid less attention to the finer points. (Much of what I had been doing was well-instilled in others; only so much time in a day; and I must choose to pick my battles.) I have yet to look at Twenty Sixteen's code. I am not sure how close the more recent default themes adhere to what in my mind are my ideals, but let me outline them below:

Q: Should braces be used, even when they are not required?
A: Braces must always be used.

Q: When can the alternative statement style (e.g. if/endif) be used?
A: In the following four situations (it's possible this isn't an exhaustive list):
1. When you are breaking out of PHP and into HTML inside the the control statement.
2. When you have an exceedingly long block and you think `endforeach` or whatever will make it easier to identify the end of the block. (Note: This often means you should refactor.)
3. When using while/endwhile for the Loop. (Note: this is often covered by A1 or A2.)
4. When wrapping a `function_exists()` or `class_exists()` check around a function or class.

Additionally, even if A1-4 do not apply, if you want to use alternative style _in a theme_, I'm not inherently against it. I would defer to whatever we seem to have done over the last six default themes. (If they have drifted away from what I'm spelling out over the years, it's certainly possible it's because I haven't paid attention.) I am however always against using braces when you're breaking out of PHP.

nacin commented:
The brace requirement (was established in late 2013)[https://make.wordpress.org/core/2013/11/13/proposed-coding-standards-change-always-require-braces/]. It should be noted that every if statement in Twenty Fifteen (and nearly every one in Twenty Fourteen) have braces. That one should be easy to enforce, along with always using the alternative style when you're breaking out of PHP into HTML.

ihorvorotnov commented:
@nacin thank you for the very (!) quick and detailed response. As you pointed out, alternative syntax is more appropriate _when you're breaking out of PHP and into HTML inside the the control statement_ (I call it mixin PHP and HTML). And in pure PHP, especially if there's only 1 function call inside the If statement, braces are better (except complex chunks of code, of course). That's what I came up with, and that's what I propose to update WPCS with, eventually.

iamtakashi commented:
In my testing, I get a parse error in archive.php, it seems like we have an extra `have_posts()` check before the pagination.

@ihorvorotnov, can you remove it and re-submit a patch after testing?

ihorvorotnov commented:
@iamtakashi that's why we need Travis CI up and running :) It would have told me about the error. Sure, will take care of that today.

ihorvorotnov commented:
@iamtakashi The error is fixed now, ready to merge.

iamtakashi commented:
@ihorvorotnov, we don't even need that `have_posts()` check L46. I think we can remove the check entirely.

ihorvorotnov commented:
@iamtakashi, right you are! I'm surprised it wasn't noticed by anyone before. Removed that check and also added a comment to the end of the loop, similar to that we have in other templates (index, etc).

iamtakashi commented:
Thank you, @ihorvorotnov!

SergeyBiryukov commented:
There was also a [good discussion](https://core.trac.wordpress.org/ticket/24549) on braces vs. `if/endif` in Core Trac, which ended up as the current wording in WPCS.

The changes in 669130f all look good to me.
